### ISSUE TITLE
Failed to create deployment with ipvs enabled in 1.8.2

### ISSUE THREAD

**Issue by gyliu513** [kubernetes/kubernetes#56054]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>

/kind bug

> /kind feature

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
1) Start kube proxy as following:
```
        "name": "proxy",
        "image": "kubernetes:v1.8.2",
        "imagePullPolicy": "IfNotPresent",
        "command": [
          "/hyperkube",
          "proxy",
          "--kubeconfig=/var/lib/kubelet/kube-proxy-config",
          "--hostname-override=9.111.255.91",
          "--feature-gates=SupportIPVSProxyMode=true",
          "--proxy-mode=ipvs",
          "--ipvs-min-sync-period=5s",
          "--ipvs-sync-period=5s",
          "--ipvs-scheduler=rr",
          "--cluster-cidr=10.1.0.0/16"
        ],
```

2) Create an deployment as following:
```
[root@gyliu-icp-1 gyliu]# cat n.yaml
apiVersion: v1
kind: Service
metadata:
  labels:
    run: my-nginx
  name: my-nginx
  namespace: default
spec:
  ports:
  - name: nginx
    port: 80
    protocol: TCP
    targetPort: 80
  selector:
    run: my-nginx
  type: NodePort
---
apiVersion: apps/v1beta1
kind: Deployment
metadata:
  labels:
    run: my-nginx
  name: my-nginx
  namespace: default
spec:
  replicas: 1
  selector:
    matchLabels:
      run: my-nginx
  template:
    metadata:
      labels:
        run: my-nginx
    spec:
      containers:
      - image: nginx:1.8.1
        imagePullPolicy: IfNotPresent
        name: my-nginx
        ports:
        - containerPort: 80
          protocol: TCP
```

3) Create the deployment and service, got error log in kubelet

```
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.alpha.kubernetes.io/notReady:NoExecute for 300s
                 node.alpha.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason                  Age                From                   Message
  ----     ------                  ----               ----                   -------
  Normal   Scheduled               21m                default-scheduler      Successfully assigned my-nginx-54d9957dd5-mbgmf to 9.111.255.91
  Normal   SuccessfulMountVolume   21m                kubelet, 9.111.255.91  MountVolume.SetUp succeeded for volume "default-token-r4xk9"
  Normal   SandboxChanged          17m (x8 over 21m)  kubelet, 9.111.255.91  Pod sandbox changed, it will be killed and re-created.
  Warning  FailedCreatePodSandBox  6m (x29 over 21m)  kubelet, 9.111.255.91  Failed create pod sandbox.
  Warning  FailedSync              1m (x38 over 21m)  kubelet, 9.111.255.91  Error syncing pod
```

```
Nov 20 08:17:15 gyliu-icp-3.novalocal hyperkube[30472]: Calico CNI checking for existing endpoint: <nil>
Nov 20 08:17:15 gyliu-icp-3.novalocal hyperkube[30472]: 2017-11-20 08:17:15.310 [INFO][2916] k8s.go 67: Extracted identifiers for CmdAddK8s Node="gyliu-icp-3.novalocal" Orchestrator="k8s" Workload="default.my-nginx-54d9957dd5-mbgmf"
Nov 20 08:17:45 gyliu-icp-3.novalocal hyperkube[30472]: E1120 08:17:45.313551   30472 cni.go:301] Error adding network: Get https://10.0.0.1:443/api/v1/namespaces/default/pods/my-nginx-54d9957dd5-mbgmf: dial tcp 10.0.0.1:443: i/o timeout
Nov 20 08:17:45 gyliu-icp-3.novalocal hyperkube[30472]: E1120 08:17:45.313574   30472 cni.go:250] Error while adding to cni network: Get https://10.0.0.1:443/api/v1/namespaces/default/pods/my-nginx-54d9957dd5-mbgmf: dial tcp 10.0.0.1:443: i/o timeout
Nov 20 08:17:45 gyliu-icp-3.novalocal hyperkube[30472]: E1120 08:17:45.443562   30472 remote_runtime.go:92] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "my-nginx-54d9957dd5-mbgmf_default" network: Get https://10.0.0.1:443/api/v1/namespaces/default/pods/my-nginx-54d9957dd5-mbgmf: dial tcp 10.0.0.1:443: i/o timeout
Nov 20 08:17:45 gyliu-icp-3.novalocal hyperkube[30472]: E1120 08:17:45.443627   30472 kuberuntime_sandbox.go:54] CreatePodSandbox for pod "my-nginx-54d9957dd5-mbgmf_default(d400586c-cdf4-11e7-beef-fa163e4a45bf)" failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "my-nginx-54d9957dd5-mbgmf_default" network: Get https://10.0.0.1:443/api/v1/namespaces/default/pods/my-nginx-54d9957dd5-mbgmf: dial tcp 10.0.0.1:443: i/o timeout
Nov 20 08:17:45 gyliu-icp-3.novalocal hyperkube[30472]: E1120 08:17:45.443642   30472 kuberuntime_manager.go:632] createPodSandbox for pod "my-nginx-54d9957dd5-mbgmf_default(d400586c-cdf4-11e7-beef-fa163e4a45bf)" failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "my-nginx-54d9957dd5-mbgmf_default" network: Get https://10.0.0.1:443/api/v1/namespaces/default/pods/my-nginx-54d9957dd5-mbgmf: dial tcp 10.0.0.1:443: i/o timeout
Nov 20 08:17:45 gyliu-icp-3.novalocal hyperkube[30472]: E1120 08:17:45.443708   30472 pod_workers.go:182] Error syncing pod d400586c-cdf4-11e7-beef-fa163e4a45bf ("my-nginx-54d9957dd5-mbgmf_default(d400586c-cdf4-11e7-beef-fa163e4a45bf)"), skipping: failed to "CreatePodSandbox" for "my-nginx-54d9957dd5-mbgmf_default(d400586c-cdf4-11e7-beef-fa163e4a45bf)" with CreatePodSandboxError: "CreatePodSandbox for pod \"my-nginx-54d9957dd5-mbgmf_default(d400586c-cdf4-11e7-beef-fa163e4a45bf)\" failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod \"my-nginx-54d9957dd5-mbgmf_default\" network: Get https://10.0.0.1:443/api/v1/namespaces/default/pods/my-nginx-54d9957dd5-mbgmf: dial tcp 10.0.0.1:443: i/o timeout"
Nov 20 08:17:45 gyliu-icp-3.novalocal hyperkube[30472]: W1120 08:17:45.533202   30472 pod_container_deletor.go:77] Container "f6357693f65981026c07c116038c0d0e39a2eea8ab94120158033cee2a37624f" not found in pod's containers
Nov 20 08:17:45 gyliu-icp-3.novalocal hyperkube[30472]: W1120 08:17:45.835335   30472 cni.go:265] CNI failed to retrieve network namespace path: Cannot find network namespace for the terminated container "f6357693f65981026c07c116038c0d0e39a2eea8ab94120158033cee2a37624f"
```

```
[root@gyliu-icp-1 gyliu]# curl 10.0.0.1:443

[root@gyliu-icp-1 gyliu]#
```
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/sig network

/cc @m1093782566

**Comment by Lion-Wei** [comment]
Similar issue happened to me before, ipvs mode `NodePort` function may not very stable before `ipset` merged in(#54219), related with node/cni and other things. Suggest use newest version. : )

**Comment by cmluciano** [comment]
This should be fixed in the 1.9 release. Please re-open if it still persists.
